File: commonpkg/models/quality-model.go
package models

type QualityDto struct {
	Id        string `json:"id"`
	Name      string `json:"name"`
	HsnCode   string `json:"hsnCode"`
	ProductId string `json:"productId"`
}

type QualityListResponse struct {
	CommonListResponse
	Data []QualityDto `json:"data,omitempty"`
}

type QualityResponse struct {
	CommonResponse
	Data QualityDto `json:"data,omitempty"`
}

type ProductDto struct {
	Id   string `json:"id"`
	Name string `json:"name"`
}

type ProductListResponse struct {
	CommonListResponse
	Data []ProductDto `json:"data,omitempty"`
}

type ProductResponse struct {
	CommonResponse
	Data ProductDto `json:"data,omitempty"`
}

File: ui/kp-textile/src/app/models/quality-model.ts
import { CommonListResponse, CommonResponse } from "./genric-model"
import { BailDetailsDto } from "./item-model";

export interface QualityDto {
	id: string
	name: string
	hsnCode: string
	productId: string
	productName: string
}
export interface QualityListItemDto extends QualityDto {
	pendingQuantity: number
	bailDetails: BailDetailsDto[]
}
export interface QualityListResponse extends CommonListResponse {
	data: QualityDto[]
}

export interface QualityResponse extends CommonResponse {
	data: QualityDto
}

export interface ProductDto {
	id: string
	name: string
}

export interface ProductListResponse extends CommonListResponse {
	data: ProductDto[]
}

export interface ProductResponse extends CommonResponse {
	data: ProductDto
}